src/test/scala/scalapb/json4s/EnumFormatSpec.scala

```diff
@@ -0,0 +1,33 @@
+package scalapb.json4s
+
+import jsontest.test.MyEnum
+import org.scalatest.{FlatSpec, MustMatchers}
+
+class EnumFormatSpec extends FlatSpec with MustMatchers with JavaAssertions {
+
+  "Enum" should "be serialized the same way as java" in {
+    assertJsonIsSameAsJava(jsontest.test.EnumTest())
+    assertJsonIsSameAsJava(jsontest.test.EnumTest(Some(MyEnum.V1)))
+  }
+
+  "Enum" should "be serialized the same way as java - non-existent string value" in {
+    val parser = JavaJsonParser.ignoringUnknownFields()
+    val enumJson = """{"enum": "XOXO"}"""
+
+    javaParse(enumJson, jsontest.Test.EnumTest.newBuilder, parser).toString must be("")
+    new Parser().ignoringUnknownFields.fromJsonString[jsontest.test.EnumTest](enumJson) must be(
+      jsontest.test.EnumTest(None)
+    )
+  }
+
+  "Enum" should "be serialized the same way as java - non-existent int value" in {
+    val parser = JavaJsonParser.ignoringUnknownFields()
+    val enumJson = """{"enum": 10}"""
+
+    javaParse(enumJson, jsontest.Test.EnumTest.newBuilder, parser).toString must be("")
```

Comment:
It looks like fact that this test pass shows a difference in behavior. The java parser lost the unrecognized value.

Reply:
A good waqy to show that you are getting the same object: https://github.com/scalapb/scalapb-json4s/blob/master/src/test/scala/scalapb/json4s/JsonFormatSpec.scala#L202-L206 is to convert enumtest from java to scala and assert equality.

Reply:
Used your advice for testing java/scala interop - [2 tests from here](https://github.com/viliusl/scalapb-json4s/blob/0fb8bc5503fd874dcafb52e826a001ed59694be9/src/test/scala/scalapb/json4s/EnumFormatSpec.scala#L41-L40)

I also extended tests with explicit proto2/proto3 variations - which led me to [adding this branch](https://github.com/viliusl/scalapb-json4s/blob/53bf40ad71848e71c3d276cf9bb7a079904a33f6/src/main/scala/scalapb/json4s/JsonFormat.scala#L393-L392)

There isssue though with Scala/Java compatability:
 - in java, (proto2/proto3), for unknown string enum value and `ignoringUnknownFields` you get value set to null;
 - in scala proto2 - you get set to `None`;
 - in scala proto3 - enum field is not `Optional`, so I cannot set it to `None`. I did set it to `0-index` value as per proto3 spec, but it's not symmetrical to java

what would be your verdict on this behavior diff?

Reply:
Can you double-check the behavior in java where you are getting nulls? I wasn't able to reproduce this with protobuf-3.8.0, and if this is true, you have probably found a bug in protobuf-java. Here is what I tried:

    import jsontest.{Test, Test3}
    val p2Builder = Test.TestAllTypes.newBuilder()
    JavaJsonParser.ignoringUnknownFields().merge("""{"optionalNestedEnum": "TTT"}""", p2Builder)
    val p2 = p2Builder.build()
    println("P2", p2)
    println("P2", p2.getOptionalNestedEnum)

    val p3Builder = Test3.MyTest3.newBuilder()
    JavaJsonParser.ignoringUnknownFields().merge("""{"optEnum": "TTT"}""", p2Builder)
    val p3 = p3Builder.build()
    println("P3", p3)
    println("P3", p3.getOptEnum)

In both cases I am getting default values.